filebeat/tests/integration/filestream_test.go

```diff
@@ -657,6 +671,471 @@ func TestFilestreamTakeOverFromLogInput(t *testing.T) {
+	testDataPath, err := filepath.Abs("./testdata")
+	if err != nil {
+		t.Fatalf("cannot get absolute path for 'testdata': %s", err)
+	}
+
+	filebeat := integration.NewBeat(
+		t,
+		"filebeat",
+		"../../filebeat.test",
+	)
+	workDir := filebeat.TempDir()
+
+	logFile := filepath.Join(workDir, "log.log")
+	logData := strings.Join(logFileLines[:5], "\n")
+	logData += "\n" // Filebeat needs the '\n' to read the last line
+	if err := os.WriteFile(logFile, []byte(logData), 0o644); err != nil {
+		t.Fatalf("cannot write log file '%s': %s", logFile, err)
+	}
+
+	fileWatcher := NewFileWatcher(t, logFile)
+	fileWatcher.SetEventCallback(func(event fsnotify.Event) {
+		if event.Has(fsnotify.Remove) {
+			t.Errorf("File %s should not have been removed, removal happened at %s",
+				event.Name,
+				time.Now().Format(time.RFC3339Nano))
+		}
+	})
+	fileWatcher.Start()
+	defer fileWatcher.Stop()
+
+	esURL := integration.GetESURL(t, "http")
+
+	// Create and start the proxy server
+	proxy := &DisablingProxy{target: &esURL, enabled: true}
+	server := &http.Server{
+		Addr:              "localhost:9201",
+		Handler:           proxy,
+		ReadHeaderTimeout: time.Second / 2,
+	}
+	go func() {
+		if err := server.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			t.Errorf("Proxy server failed: %s", err)
+		}
+	}()
+	defer server.Close()
+
+	proxyURL, err := url.Parse(server.Addr)
+	if err != nil {
+		t.Fatalf("cannot parse proxy URL: %s", err)
+	}
+
+	user := esURL.User.Username()
+	pass, _ := esURL.User.Password()
+	vars := map[string]any{
+		"homePath":    workDir,
+		"logfile":     logFile,
+		"testdata":    testDataPath,
+		"esHost":      proxyURL.String(),
+		"user":        user,
+		"pass":        pass,
+		"index":       index,
+		"gracePeriod": gracePeriod.String(),
+	}
+
+	cfgYAML := getConfig(t, vars, "delete", "real-es.yml")
+	filebeat.WriteConfigFile(cfgYAML)
+	filebeat.Start()
+
+	// Wait for data in ES
+	msgs := []string{}
+	require.Eventually(t, func() bool {
+		msgs = getEventsMsgFromES(t, index, 200)
+		return len(msgs) == len(logFileLines)/2
+	}, time.Second*10, time.Millisecond*100, "not all log messages have been found on ES")
+
+	// Wait for 1/2 of the grace period and add more data
+	time.Sleep(gracePeriod / 2)
+
+	// Add more data to the file
+	f, err := os.OpenFile(logFile, os.O_RDWR|os.O_APPEND, 0644)
+	if err != nil {
+		t.Fatalf("cannot open logfile to append data: %s", err)
+	}
+	logData2 := strings.Join(logFileLines[5:], "\n")
+	logData2 += "\n"
+	if _, err := f.WriteString(logData2); err != nil {
+		t.Fatalf("could not append data to log file: %s", err)
+	}
+	if err := f.Sync(); err != nil {
+		t.Fatalf("cannot flush log file: %s", err)
+	}
+	if err := f.Close(); err != nil {
+		t.Fatalf("cannot close log file: %s", err)
+	}
+
+	// Disable (aka block) the output
+	proxy.Disable()
+
+	// Wait twice the grace period before unblocking the output
+	blockedTimer := time.NewTimer(gracePeriod * 2)
+	<-blockedTimer.C
```

Comment:
This block makes it so we don't test that appending data to the file extends the grace period, because we block the output at the same time making this primarily a test that deletion is delayed until all events are acknowledged.

Reply:
> If I'm reading this correctly, you'd like me to split the output blocked from the data added during the grace period, is that it?

There needs to be a test explicitly showing that writing data to a file during the grace period resets the grace period. This is how I expect it to work.

Otherwise the grace period is a one time thing counted from the first time you reached EOF which is a lot harder to reason about.

Reply:
I see what you're saying, I'll make a more explicit test for this, without mixing it with the output blocked.